### ISSUE TITLE
Confused by zeroMQ

### ISSUE THREAD

**Issue by bentalentville** [ratchetphp/Ratchet#1073]
I got the basic websocket code (from the http://socketo.me/docs/push page) to work just fine, but I got completely confused when zeroMQ was added in.  Is the pusher code meant to be IN ADDITION to the socket code that I have working for the basic client communication?  Or does it replace it?  Feels like it is in addition which means the socket app would still be run for client comm and the pusher would loop around to allow my back end PHP code to communicate with my users, triggered via various AJAX requests.  Is that correct?  I would need to start both of them before running anything in a browser?

It could be that the zeroMQ code supercedes the simple websocket code, can anyone shed some light on it so I can figure out how my back end code can initiate data pushes to desired clients/users?

**Comment by PaulRotmann** [comment]
You're correct in your understanding! The ZeroMQ pusher code is in addition to your basic WebSocket server, not a replacement.

**Architecture Overview:**
1. **WebSocket Server** - Handles client connections (browser ↔ server)
2. **ZeroMQ Pusher** - Allows backend PHP to push data to connected clients
3. **Your Backend** - Triggers pushes via AJAX/HTTP requests

**How it works:**
- Your WebSocket server runs continuously, managing client connections
- ZeroMQ creates a bridge between your backend PHP and the WebSocket server
- When your backend needs to push data, it sends a message through ZeroMQ
- The WebSocket server receives the ZeroMQ message and forwards it to specific clients

**Yes, you need both running:**
- Start your WebSocket server (handles client connections)
- Your backend code uses ZeroMQ to send messages to the WebSocket server
- The WebSocket server then pushes those messages to the appropriate clients

This pattern lets your regular PHP application (triggered by AJAX, forms, etc.) push real-time updates to connected WebSocket clients.

We're giving Ratchet some much-needed love and attention! As part of our issue cleanup initiative (#1100), we'd love to continue this discussion in our GitHub Discussions where we can provide more detailed architecture examples and help you implement this pattern. Please create a new discussion post and reference this issue.

If you want to support Ratchet, please consider [sponsoring our work](https://github.com/sponsors/reactphp)! ❤️
